From 1e17eecefb7477e4cd7a3a9887323a3d649db174 Mon Sep 17 00:00:00 2001
From: Luo Jie <quic_luoj@quicinc.com>
Subject: arm64: dts: qcom: ipq5332: Add MDIO device tree
Date: Wed, 10 Jan 2024 19:20:56 +0800

Add the MDIO device tree of ipq5332.

Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
[Til Kaiser: drop board-specific pinctrl properties from the generic mdio node]
Signed-off-by: Til Kaiser <mail@tk154.de>
---
 arch/arm64/boot/dts/qcom/ipq5332.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq5332.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq5332.dtsi
@@ -803,6 +803,16 @@
 				ranges;
 			};
 		};
+
+		mdio: mdio@90000 {
+			compatible = "qcom,ipq4019-mdio";
+			reg = <0x90000 0x64>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			clocks = <&gcc GCC_MDIO_AHB_CLK>;
+			clock-names = "gcc_mdio_ahb_clk";
+			status = "disabled";
+		};
 	};
 
 	thermal-zones {
